src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -164,7 +166,18 @@ private function translateUsingLegacySystem($message, array $parameters, $domain
         $sourceFile = (!empty($domainParts[2])) ? strtolower($domainParts[2]) : $moduleName;
 
         // translate using the legacy system WITHOUT fallback to the new system (to avoid infinite loop)
-        return (new LegacyTranslator())->translate($moduleName, $message, $sourceFile, $parameters, false, $locale, false);
+        $translated = (new LegacyTranslator())->translate($moduleName, $message, $sourceFile, $parameters, false, $locale, false);
+
+        //Translate::getModuleTranslation performs a htmlspecialchars by default, which we want to cancel in this case
+        //Note: this is only cases where l is called with a domain but the key was not found
+        if (null === $legacy) {
+            $translated = htmlspecialchars_decode($translated);
+        } else if ('htmlspecialchars' !== $legacy) {
+            $translated = htmlspecialchars_decode($translated);
+            $translated = call_user_func($legacy, $translated);
+        }
```

Comment:
I'm not that happy with the idea of applying `htmlspecialchars` and then reversing it, it's hacky.

I think the correct answer would be to refactor the code from `Translate::getModuleTranslation` in `LegacyTranslator` so that `htmlspecialchars` never happens when it's called from new code.